### DISCUSSION TITLE
Raspberry Pi unstable MIDI ID

### DISCUSSION THREAD

**Discussion by newVideophile** [GrandOrgue/grandorgue#1616] [Help] [answered]
I recently updated my Raspberry Pi. It has a convenient tool that upgrades the os and all installed software.

After the update,  I had a new issue: the MIDI configuration would change randomly after reboot. I.e. my keyboards and pistons did not work any more. I used right clic + "wait for MIDI event" to set everything properly again, but after reboot, the configuration would jump to another MIDI ID number. A power off (rather than a software reboot) seemed to maintain the ID value.

 
![MIDI_issue_RPi](https://github.com/GrandOrgue/grandorgue/assets/95248680/0fe6edce-1abd-4ab6-a2cd-dadd32f343f7)

Any idea ?

I downgraded back my Raspberry Pi to the previous version, and it now works well, but in the long run, this will be a problem.

**Comment by oleg68** [comment]
First of all, you should set up your midi devices at File->Settings->Midi Devices

**Comment by newVideophile** [comment]
Oleg, I don't get your point. My problem is not that my devices are not set up, but that in somes cases, their setup does not survive to a reboot. 

By the way, I suspect that the problem is caused by the Rapsberry Pi rather than GO.

**Reply by oleg68** [reply]
You can set up wildcard patterns for matching your devices at File->Settings->Midi devices

**Comment by newVideophile** [comment]
How to do that ? In the Midi devices menu, I only see checkboxes

![MIDI_devices](https://github.com/GrandOrgue/grandorgue/assets/95248680/45ecaef0-11fa-4927-9c9c-e937b295a87f)

**Reply by oleg68** [reply, marked as answer]
You have a very old version of GrandOrgue. Please upgrade to the latest one

**Reply by newVideophile** [reply]
Yes, because my previous attempts to upgrade GO created other issues. I don't remember what went wrong, but it was just unusable.

Anyway, I will try to update both the raspberry pi and and GO and see what happens

**Reply by newVideophile** [reply]
Ok, GO upgrade done. 

After installation, I had a weird effect: the reverb was working only in the right stereo channel. I changed the audio output from "RtAudio" to "PortAudio" and bug was cured. 

And most importantly, my initial issue is solved. I even didn't have to use  wildcard patterns. Thank you!

**Reply by newVideophile** [reply]
After more tests, the problem is still solved (I actually needed the wildcard patterns), but a new one arised :(

I have the same kind of issue with the audio device ID:
![audio_interface](https://github.com/GrandOrgue/grandorgue/assets/95248680/5097f22f-b38f-4d1d-b585-8d54174b68bd)

Either I use the first item in the list, but it will randomly switch between hw(2;0) and hw(3;0) after reboot, or I use the last item in the list, but then I have the "reverb-in-one-side-only" bug.

**Comment by oleg68** [comment]
Unfortunally, there is no audio matching capability in GrandOrgue yet: See #1265 for details.

But there is a workaround: setup `Jack` or `Pipewire` with jack interface.

**Reply by newVideophile** [reply]
I just figured out that I just need to select the default audio device, and now everything is working as expected. 

(I still have an issue with the CPU governor, related to the OS update, but this has nothing to do with GO, of course)
